homework-submission/abdullahReact/main.js

```diff
@@ -0,0 +1,98 @@
+const toDoItems = [
+  {
+    id: 1,
+    description: 'Get out of bed',
+    deadline: '2017-09-11',
+    done: true,
+  },
+  {
+    id: 2,
+    description: 'Brush teeth',
+    deadline: '2017-09-10',
+    done: false,
+  },
+  {
+    id: 3,
+    description: 'Eat breakfast',
+    deadline: '2017-09-09',
+    done: false,
+  },
+];
+
+//the goal of this component is to define/create a reusable list item that uses own function to set state.
+// in this case to set as done or not done by clicking
+
+class ListItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { done: props.done };
+    this.switchDoneState = this.switchDoneState.bind(this);
+  }
+
+  switchDoneState() {
```

Comment:
Cool that you can check and uncheck a todo item! Did you notice that when you click on an item in the static list that it doesn't get a line-through straight away? Do you know why?

Reply:
I figured that out. 
when I call LitstItem inside Staticlist, I didn't pass the "done" property to write in state. 
that's why it tries to assign, but because of there is nothing it returns undefined. 
than "I guess" when I click, it takes the opposite of undefined it takes it like true. 
(by the way, the style "line-trough" applied when its false.) I corrected both the style and props.

Reply:
Ha super! Must have been quite a tricky one to figure out, I didn't see it on first sight. Cool! :-)

Reply:
yes it was hard to see. wee looked that with my friend Talip in class. he noticed that.